spec/requests/api/querying_spec.rb

```diff
@@ -323,9 +323,21 @@ def create_vms_by_name(names)
 
       run_get vms_url, :expand => "resources", :attributes => "name,vendor"
 
-      expect_query_result(:vms, 1, 1)
```

Comment:
>  Perhaps there should be a separate test for structure versus content?

@Fryguy would it be possible to test the content without having to also specify its structure? Perhaps you can elaborate.

My feeling (after some thought) is that I'm not averse to this *structural* duplication in tests - I don't think it's the same thing as trying to encapsulate the behavior of some thing in a system. That kind of duplication of *knowledge* isn't just painful because it means having to go to more places to update things, but because it adds complexity to the system as a whole.

So I think the downside to introducing structural duplication across tests is an increased cost of change if say, we decide to modify the way we represent count/subcount in the response JSON (and we may well do). But that cost is relatively small, and easy to manage with our modern, powerful text editors. But it is a cost, nonetheless.

I think that cost is more than offset by the clarity of seeing the structure and content of an expectation for one specific response in the body of the test as a single literal (in this case a hash). But I know opinions vary.

So in general I prefer having a single literal. We could implement something like:

```ruby
expected = {}
expected.merge(query_result(:name => name, :count => count, :subcount => subcount))
```

But as you can see, I've introduced incidental duplication in the structure of the method signature (which will do nothing more than return its argument, and I'd prefer named arguments here for clarity), and increased indirection, for little gain.

I could simply leave the name/count/subcount checking to the method above, it was just my preference to check the response body once in one place per test. Now that we've merged failure aggregation for API specs, this doesn't seem to be as much of a problem as it was for me previously (though it doesn't seem to work perfectly, I need to do more research here).

@Fryguy any thoughts on the above?